src/execution/perfect_aggregate_hashtable.cpp

```diff
@@ -110,15 +111,29 @@ void PerfectAggregateHashTable::AddChunk(DataChunk &groups, DataChunk &payload)
 
 	// after finding the group location we update the aggregates
 	idx_t payload_idx = 0;
-	for (idx_t aggr_idx = 0; aggr_idx < aggregates.size(); aggr_idx++) {
-		auto &aggr = aggregates[aggr_idx];
-		auto input_count = (idx_t)aggr.child_count;
-		aggr.function.update(input_count == 0 ? nullptr : &payload.data[payload_idx], input_count, addresses,
-		                     payload.size());
+	for (auto &aggregate : aggregates) {
+		auto input_count = (idx_t)aggregate.child_count;
+		if (aggregate.filter) {
+			ExpressionExecutor filter_execution(aggregate.filter);
+			SelectionVector true_sel(STANDARD_VECTOR_SIZE);
```

Comment:
This seems like the exact same code as the regular AggregateHashtable. I would unify it with that code by using a static function (`AggregateHashtable::UpdateAggregate(...)`).